frontend/src/metabase/admin/databases/containers/DatabaseEditApp.jsx

```diff
@@ -204,7 +223,17 @@ class DatabaseEditApp extends Component {
   }
 }
 
-export default _.compose(
+const getSubmitError = error => {
+  if (isObject(error?.data?.errors)) {
```

Comment:
Without this line, [this test](https://github.com/metabase/metabase/blob/6ee13aa266ec2b8a95f0b98b10fda0cf8a3c4f15/frontend/test/metabase/scenarios/admin/databases/add.cy.spec.js#L141) would fail.

This, however, means this snippet is now different from its origin from where it's copied.
https://github.com/metabase/metabase/blob/62885b4084b7b449c77ed5fd39fd3bce5c365afc/frontend/src/metabase/setup/components/DatabaseStep/DatabaseStep.tsx#L202-L206

I considered whether I shouldn't have this guard check and change the API response in that test to reflect the actual API response instead. But at the same time crashing the form with a simple text response sounds odd, that's why I decided to have a guard check here.